proposals/0000-splitting-out-stable-interfaces-from-th.rst

```diff
@@ -0,0 +1,335 @@
+Each new version of GHC in practice comes with a new major version of both of these packages.
+Therefore, a strict upper-bound guarantees that end-users don't accidentally build the library against a version of GHC untested by the maintainer.
+When the maintainer is satisfied that the new version of GHC doesn't break the library, they can release a Hackage revision with updated bounds.
+
+The direction of travel in this proposal and efforts to make both ``base`` and ``template-haskell`` decoupled from GHC versions, make this strategy more difficult to implement.
+If a maintainer replaces a dependency on ``template-haskell`` by one on ``template-haskell-lift`` and ``base`` becomes reinstallable,
+then they can no longer depend on a new version of GHC forcing an upgrade of one of these packages.
+So, their library might be built by an end-user with a version of GHC the maintainer hasn't yet tested.
+
+We could argue that the maintainer doesn't have a reason to be worried since GHC comes with strong `stability principles <../principles.rst>`_. Yet, these also have exceptions.
+
+Consider the following scenario:
+1. GHC-9.14 adds support to ``DeriveLift`` for ``GADT``\s.
+2. The maintainer adds ``DeriveLift`` instances for their ``GADT``\s.
+3. The GHC maintainers realize that there is a fundamental issue with this new feature and that it must be reverted in GHC-9.16.
+4. Now the maintainers code is broken by the release of GHC-9.16
+
+A way to defend against this would be to count changes to ``DeriveLift`` as part of the interface ``template-haskell-lift``.
+Then whenever the class of datatypes supported by ``DeriveLift`` changed, we would make a new major version of ``template-haskell-lift``.
+
+While we hold these to be valid concerns, we do not think expanding the interface of ``template-haskell-lift`` is the way to resolve them.
+This suggestion goes against our aim of decoupling ``template-haskell-lift`` from GHC.
+It is possible to shim over changes to the ``Lift`` interface in ``ghc-internal`` as we've outlined in :ref:`independence`.
+But it is not possible to shim over features implemented in the compiler.
+If the implementation of ``DeriveLift`` changed then we would be forced to have a sequence of versions of ``template-haskell-lift`` without either backwards- or forwards-compatibility.
+This would greatly limit the benefits of this proposal.
+
+We feel that it is reasonable that a library maintainer should be able to explicitly define which versions of GHC they support, but that this can be done orthogonally to this proposal.
+In fact, tackling this independently might lead to a more general solution.
+For instance, we could have a ``ghc-version`` package, which exports nothing but carries the version of GHC. We would have to allow it to be exempted from unused package warnings, but this seems doable.
```

Comment:
(Non-actionable as far as the proposal goes.) 

Why isn't `ghc-version` simply `ghc-internal`? That is, we could just ensure that `ghc-internal` is versioned according to the GHC version it comes with. Which incidentally is the case *already*. So `ghc-internal` is `ghc-version`.